src/test/java/net/snowflake/client/jdbc/DatabaseMetaDataIT.java

```diff
@@ -139,41 +140,50 @@ public void testGetSchemas() throws Throwable {
+      String currentSchema = connection.getSchema();
       assertEquals("schema", metaData.getSchemaTerm());
-      ResultSet resultSet = metaData.getSchemas();
-      verifyResultSetMetaDataColumns(resultSet, DBMetadataResultSetMetadata.GET_SCHEMAS);
-
       Set<String> schemas = new HashSet<>();
-      while (resultSet.next()) {
-        schemas.add(resultSet.getString(1));
+      try (ResultSet resultSet = metaData.getSchemas()) {
+        verifyResultSetMetaDataColumns(resultSet, DBMetadataResultSetMetadata.GET_SCHEMAS);
+        while (resultSet.next()) {
+          String schema = resultSet.getString(1);
+          if (currentSchema.equals(schema) || !TestUtil.isSchemaGeneratedInTests(schema)) {
+            schemas.add(schema);
+          }
+        }
       }
-      resultSet.close();
-      assertThat(schemas.size(), greaterThanOrEqualTo(1)); // one or more schemas
+      assertThat(schemas.size(), greaterThanOrEqualTo(1));
 
       Set<String> schemasInDb = new HashSet<>();
-      resultSet = metaData.getSchemas(connection.getCatalog(), "%");
-      while (resultSet.next()) {
-        schemasInDb.add(resultSet.getString(1));
+      try (ResultSet resultSet = metaData.getSchemas(connection.getCatalog(), "%")) {
+        while (resultSet.next()) {
+          String schema = resultSet.getString(1);
+          if (currentSchema.equals(schema) || !TestUtil.isSchemaGeneratedInTests(schema)) {
+            schemasInDb.add(schema);
+          }
+        }
       }
-      assertThat(schemasInDb.size(), greaterThanOrEqualTo(1)); // one or more schemas
+      assertThat(schemasInDb.size(), greaterThanOrEqualTo(1));
       assertThat(schemas.size(), greaterThanOrEqualTo(schemasInDb.size()));
-      assertTrue(schemas.containsAll(schemasInDb));
-      assertTrue(schemas.contains(connection.getSchema()));
+      schemasInDb.forEach(schemaInDb -> assertThat(schemas, hasItem(schemaInDb)));
+      assertTrue(schemas.contains(currentSchema));
+      assertTrue(schemasInDb.contains(currentSchema));
     }
 
     // CLIENT_METADATA_REQUEST_USE_CONNECTION_CTX = true
-    try (Connection connection = getConnection()) {
-      Statement statement = connection.createStatement();
+    try (Connection connection = getConnection();
+        Statement statement = connection.createStatement()) {
       statement.execute("alter SESSION set CLIENT_METADATA_REQUEST_USE_CONNECTION_CTX=true");
 
       DatabaseMetaData metaData = connection.getMetaData();
       assertEquals("schema", metaData.getSchemaTerm());
-      ResultSet resultSet = metaData.getSchemas();
-      Set<String> schemas = new HashSet<>();
-      while (resultSet.next()) {
-        schemas.add(resultSet.getString(1));
+      try (ResultSet resultSet = metaData.getSchemas()) {
+        Set<String> schemas = new HashSet<>();
+        while (resultSet.next()) {
+          schemas.add(resultSet.getString(1));
+        }
+        assertThat(schemas.size(), equalTo(1)); // more than 1 schema
```

Comment:
shouldn't it be `greaterThan(1)` according to comment?

Reply:
I haven't changed this assertion - it looks like the comment is misleading so I removed the commend